homeassistant/components/ring/number.py

```diff
@@ -0,0 +1,166 @@
+        for description in NUMBER_TYPES
+        for device in ring_data.devices.all_devices
+        if description.exists_fn(device)
+    ]
+
+    async_add_entities(entities)
+
+
+class RingNumber(RingEntity[RingDeviceT], NumberEntity):
+    """A number implementation for Ring device."""
+
+    entity_description: RingNumberEntityDescription[RingDeviceT]
+
+    def __init__(
+        self,
+        device: RingDeviceT,
+        coordinator: RingDataCoordinator,
+        description: RingNumberEntityDescription[RingDeviceT],
+    ) -> None:
+        """Initialize a number for Ring device."""
+        super().__init__(device, coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{device.id}-{description.key}"
+        self._no_updates_until = time.monotonic()
+        self._update_native_value()
+
+    def _update_native_value(self) -> None:
+        # History values can drop off the last 10 events so only update
+        # the value if it's not None
+        native_value = self.entity_description.value_fn(self._device)
+        if native_value is not None:
+            self._attr_native_value = float(native_value)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Call update method."""
+
+        if self._no_updates_until > time.monotonic():
+            return
+
+        self._device = cast(
+            RingDeviceT,
+            self._get_coordinator_data().get_device(self._device.device_api_id),
+        )
+
+        self._update_native_value()
+
+        super()._handle_coordinator_update()
+
+    @exception_wrap
+    async def async_set_native_value(self, value: float) -> None:
+        """Call setter on Ring device."""
+        async_setter = getattr(self._device, f"async_set_{self.entity_description.key}")
+        await async_setter(int(value))
+
+        self._no_updates_until = time.monotonic() + SKIP_UPDATES_DELAY_SECONDS
+
+        self._attr_native_value = value
+        self.async_write_ha_state()
+
+
+@dataclass(frozen=True, kw_only=True)
+class RingNumberEntityDescription(NumberEntityDescription, Generic[RingDeviceT]):
+    """Describes Ring number entity."""
+
+    value_fn: Callable[[RingDeviceT], StateType]
+    setter_fn: Callable[[RingDeviceT, float], Awaitable[None]]
+    exists_fn: Callable[[RingGeneric], bool]
+
+
+NUMBER_TYPES: tuple[RingNumberEntityDescription[Any], ...] = (
+    RingNumberEntityDescription[RingChime](
+        key="volume",
+        translation_key="volume",
+        mode=NumberMode.SLIDER,
+        native_min_value=ring_doorbell.const.CHIME_VOL_MIN,
+        native_max_value=ring_doorbell.const.CHIME_VOL_MAX,
+        native_step=1,
+        value_fn=lambda device: device.volume,
+        setter_fn=lambda device, value: device.async_set_volume(int(value)),
+        exists_fn=lambda device: isinstance(device, RingChime),
+    ),
+    RingNumberEntityDescription[RingDoorBell](
+        key="volume",
```

Comment:
It would be great if you'd submit a PR to the library as I don't have a chime myself to test with (and it'll be tiny). It might also encourage you to add more amazing features to the lib 😄

Reply:
Haha, fair enough. Here you go: https://github.com/python-ring-doorbell/python-ring-doorbell/pull/439